File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO3/BridgePattern/src/ShapeCircle.java

import java.awt.Color;
import java.awt.Graphics;

public class ShapeCircle extends ShapeBase
{
    private int cx,cy,r;
    public ShapeCircle(String text)
    {
        parse(text, (v)->cx=v, (v)->cy=v, (v)->r=v);
    }
    
    @Override public String toXml()
    {
        return String.format("<Circle cx=\"%d\" cy=\"%d\" r=\"%d\" />", cx,cy,r);
    }
    
    @Override public void draw(Graphics g)
    {
        g.setColor(Color.red);
        g.drawOval(cx-r,cy-r,2*r,2*r);
    }

    @Override public String toString()
    {
        return String.format("Circle(cx:%d, cy:%d, r:%d)", cx,cy,r);
    }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/ShapeCircle.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    class ShapeCircle : ShapeBase
    {
        private int cx,cy,r;
        public ShapeCircle(string parameters)
        {
            Parse(parameters, (v) => cx = v, (v) => cy = v, (v) => r = v);
        }
        public override void Draw(Graphics g)
        {
            g.DrawEllipse(Pens.Red, cx - r, cy - r, 2 * r, 2 * r);
        }

        public override string ToXml()
        {
            return $"<circle cx=\"{cx}\" cy=\"{cy}\" r=\"{r}\" />";
        }
        public override string ToString()
        {
            return $"Circle(cx:{cx},cy:{cy},r:{r})";
        }
    }
}
